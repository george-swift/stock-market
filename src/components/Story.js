import PropTypes from 'prop-types';

const Story = ({
  symbol,
  title,
  text,
  site,
}) => (
  <div className="card">
    <div className="card-header">{symbol}</div>
    <div className="card-body">
      <blockquote className="blockquote mb-0">
        <h5>{title}</h5>
        <p>{text}</p>
        <footer className="blockquote-footer">
          <cite title={site}>{site}</cite>
        </footer>
      </blockquote>
    </div>
  </div>
);

Story.propTypes = {
  symbol: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  site: PropTypes.string.isRequired,
};

export default Story;
