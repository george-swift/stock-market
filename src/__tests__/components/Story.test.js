import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { render, screen, within } from '@testing-library/react';
import Story from '../../components/Story';

const symbol = 'SQ';
const title = 'News';
const text = 'Stub text';
const site = 'Test reporters';

describe('Snapshot of the Story component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Story symbol={symbol} title={title} text={text} site={site} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Querying the Story component', () => {
  beforeEach(() => render(
    <Story symbol={symbol} title={title} text={text} site={site} />,
  ));

  it('should render given news in a card', () => {
    const newsCard = document.querySelector('.card');
    expect(newsCard).toBeInTheDocument();
    expect(newsCard).not.toBeEmptyDOMElement();
  });

  it('should provide details for each news report', () => {
    const companySymbol = screen.getByText(symbol);
    expect(companySymbol).toBeInTheDocument();
    const report = document.querySelector('blockquote');
    const title = within(report).getByText('News');
    expect(title).toBeInTheDocument();
    const text = within(report).getByText('Stub text');
    expect(text).toBeInTheDocument();
    const source = within(report).getByText(site);
    expect(source).toBeInTheDocument();
  });
});
