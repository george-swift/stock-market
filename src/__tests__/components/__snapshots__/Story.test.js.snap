// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the Story component should render correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    SQ
  </div>
  <div
    className="card-body"
  >
    <blockquote
      className="blockquote mb-0"
    >
      <h5>
        News
      </h5>
      <p>
        Stub text
      </p>
      <footer
        className="blockquote-footer"
      >
        <cite
          title="Test reporters"
        >
          Test reporters
        </cite>
      </footer>
    </blockquote>
  </div>
</div>
`;
